matchms/Scores.py

```diff
@@ -199,6 +217,55 @@ def scores_by_query(self, query: QueriesType, sort: bool = False) -> numpy.ndarr
                             self._scores[references_idx_sorted, selected_idx].copy()))
         return list(zip(self.references, self._scores[:, selected_idx].copy()))
 
+    @classmethod
+    def import_from_json(cls, file_path: str) -> Scores:
+        """Import scores from a JSON file.
+
+        Parameters
+        ----------
+        file_path
+            Path to the scores file.
+        """
+        return ScoresBuilder().from_json(file_path).build()
+
+    @classmethod
+    def import_from_pickle(cls, file_path: str) -> Scores:
+        """Import scores from a pickle file.
+
+        WARNING: Pickle files are not secure and may execute malicious code. Make sure that you are importing a pickle
+        file from a trusted source.
+        """
+        with open(file_path, "rb") as f:
+            return pickle.load(f)
+
+    def export_to_file(self, filename: str, file_format: str = "json"):
+        """Export the scores to a file.
+
+        Parameters
+        ----------
+        filename
+            Path to file to write to
+        file_format
+            File format to write to. Supported formats are: "json", "pkl". Default is "json".
+        """
+        if file_format == "json":
+            with open(filename, "w", encoding="utf-8") as f:
+                json.dump(self, f, cls=ScoresJSONEncoder)
+        elif file_format == "pkl":
+            with open(filename, "wb") as f:
+                pickle.dump(self, f)
+        else:
+            raise ValueError(f"File format '{file_format}' is not supported.")
+
+    def to_dict(self) -> dict:
+        """Return a dictionary representation of scores."""
+        return {"__Scores__": True,
+                "similarity_function": self.similarity_function.to_dict(),
+                "is_symmetric": self.is_symmetric,
+                "references": [reference.to_dict() for reference in self.references],
+                "queries": [query.to_dict() for query in self.queries] if not self.is_symmetric else None,
+                "scores": self.scores.tolist()}
```

Comment:
This will work for small and medium `Scores` objects, so fine for now I would say.
But I expect this to be slow for huge arrays.

More importantly, as soon as we (finally...) manage to move to sparse arrays this will need substantial changes.
We already know that this step will break things, and it will break here, too (Scores saved with the here presented version will not import in a future version of matchms with sparse representations).

Reply:
Well, quick afterthought: We could of course implement a `.tolist()` method in our sparse scores array...

Reply:
True - but I think this is something we can address when we get to this point.

Reply:
True, was more a note to myself... :)